### DISCUSSION TITLE
Modifying and saving descriptions in Types

### DISCUSSION THREAD

**Discussion by grndng** [IfcOpenShell/IfcOpenShell#4121] [Q&A]
Hi there. I'm currently working with an IFC file that has descriptions in e.g. `IfcWallType`. According to the documentation of the schema on standards.buildingsmart (https://standards.buildingsmart.org/IFC/RELEASE/IFC4/FINAL/HTML/annex/annex-d/common-use-definitions/ifcwalltype.htm), it's the third entry in `IfcWallType`.

When I read the wall types, one of my entries looks like this:
`#613=IfcWallType('3OuTsMKBHCvvF5gjNtI_Qf',#12,'Wand Type','Type Description',$,(#623),$,'D8E1DD96-50B4-4CE7-93C5-AAD5F74BE6A9',$,.NOTDEFINED.)`

What I want to change is the entry for `"Type Description"` to e.g. `"This is the wall description"` and I couldn't find a function/api call in the documentation to do so. It turns out that the term `description` is difficult to look up because it's occuring on many pages in the documentation. When going through the Python api documentation manually, I wasn't too successful either.

**Comment by aothms** [comment]
In it's simplest form you could do something like this:

```python
f = ifcopenshell.open(...)
for inst in f.by_type('IfcWallType'):
  inst.Description = "This is the wall description"
f.write('updated-model.ifc')
```
